pgmax/fg/graph.py

```diff
@@ -196,23 +200,35 @@ def _register_factor_group(
                 f"A factor group with the name {name} already exists. Please choose a different name!"
             )
 
-        for variables, factor in factor_group._variables_to_factors.items():
-            if variables in self._variables_to_factors:
+        # TODO: do we want to remove this for speed?
```

Comment:
We probably don't want to remove. I added this check after accidentially adding the same factor twice and getting really confused about the results. I think it's a useful check....